cli/flox-watchdog/src/process.rs

```diff
@@ -0,0 +1,139 @@
+use std::fs::read_to_string;
+
+use tracing::debug;
+
+/// https://man7.org/linux/man-pages/man5/proc_pid_stat.5.html
+/// Process has finished but the parent hasn't reaped it yet.
+const PROCFS_STATE_ZOMBIE: &str = "Z";
+/// Process is dead and will transition to a zombie or disappear.
+/// Technically we shouldn't see this, but just in case:
+/// https://unix.stackexchange.com/a/653370
+const PROCFS_STATE_DEAD: &str = "X";
+const PROCFS_STATE_DEAD_COMPAT: &str = "x";
+
+/// Process watcher for Linux based on `procfs`.
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct ProcfsWatcher {
+    pid: i32,
+}
+
+/// Allow compilation when not on Linux to make IDE/LSP development easier. It
+/// will always return negative results when not used on other platforms.
+#[cfg_attr(not(target_os = "linux"), allow(dead_code))]
```

Comment:
question: why `not(target_os = "linux")` shouldnt it be available _only_ on linux?

Reply:
I want to allow dead code when not on Linux. Since we only develop on MacOS and Linux it could be:
```suggestion
#[cfg_attr(target_os = "macos", allow(dead_code))]
```

(realised there's a typo in the comment which should read "when ~~not~~ used")

I used `not(target_os = "linux"), ignore` rather than `target_os = "macos", ignore` in the tests because I kept reading the positive+negative the wrong way round but happy to change it back.

Reply:
Removed